### ISSUE TITLE
Port conflict when using custom predictor with logger

### ISSUE THREAD

**Issue by orezahc** [kubeflow/kfserving#759]
/kind bug

**What steps did you take and what happened:**
Custom Model will not works when logger is enabled.

custom.yaml
```yaml
apiVersion: serving.kubeflow.org/v1alpha2
kind: InferenceService
metadata:
  labels:
    controller-tools.k8s.io: "1.0"
  name: custom-sample
  namespace: models
spec:
  default:
    predictor:
      minReplicas: 1
      logger:
        url: http://message-dumper.default/
        mode: all
      custom:
        container:
          image: ${REGISTRY}/custom-sample
          env:
            - name: GREETING_TARGET
              value: "Python KFServing Sample"
```
message-dumper.yaml
```yaml
apiVersion: serving.knative.dev/v1alpha1
kind: Service
metadata:
  name: message-dumper
spec:
  template:
    spec:
      containers:
      - image: gcr.io/knative-releases/github.com/knative/eventing-sources/cmd/event_display
```

**What did you expect to happen:**
The custom model with logger enabled should be deployed successfully.

**Anything else you would like to add:**
The port of user-container is changed to 8081 from 8080 when logger is enabled
```
$ k logs pod/custom-sample-predictor-default-gv4qn-deployment-5b6b686f8pscwz user-container -n models
[2020-04-02 20:08:33 +0000] [1] [INFO] Starting gunicorn 20.0.4
[2020-04-02 20:08:33 +0000] [1] [INFO] Listening at: http://0.0.0.0:8081 (1)
[2020-04-02 20:08:33 +0000] [1] [INFO] Using worker: threads
[2020-04-02 20:08:33 +0000] [8] [INFO] Booting worker with pid: 8
```

```
$ k logs pod/custom-sample-predictor-default-gv4qn-deployment-5b6b686f8pscwz inferenceservice-logger -n models
{"level":"info","ts":1585858129.4851239,"logger":"entrypoint","msg":"Starting the log dispatcher"}
{"level":"info","ts":1585858129.4852054,"logger":"entrypoint","msg":"Starting","worker":1}
{"level":"info","ts":1585858129.4852393,"logger":"entrypoint","msg":"Starting","worker":2}
{"level":"info","ts":1585858129.4852538,"logger":"entrypoint","msg":"Starting","worker":3}
{"level":"info","ts":1585858129.4852717,"logger":"entrypoint","msg":"Starting","worker":4}
{"level":"info","ts":1585858129.4852853,"logger":"entrypoint","msg":"Starting","worker":5}
{"level":"info","ts":1585858129.4853013,"logger":"entrypoint","msg":"Starting","port":"8081"}
{"level":"error","ts":1585858129.4854178,"logger":"entrypoint","msg":"Failed to run HTTP server","error":"default server failed: listen tcp :8081: bind: address already in use","errorVerbose":"listen tcp :8081: bind: address already in use\ndefault server failed\nmain.main.func1\n\t/go/src/github.com/kubeflow/kfserving/cmd/logger/main.go:85\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:2361","stacktrace":"github.com/kubeflow/kfserving/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/src/github.com/kubeflow/kfserving/vendor/github.com/go-logr/zapr/zapr.go:128\nmain.main\n\t/go/src/github.com/kubeflow/kfserving/cmd/logger/main.go:93\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:198"}
```

**Environment:**

- Istio Version: `1.5.1`
- Knative Version: `0.13.2`
- KFServing Version: `0.3.0`
- Kubeflow version: N/A
- Minikube version: N/A
- Kubernetes version: (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T23:41:55Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.10", GitCommit:"150f36044fe31bee3891b5a3fae69c17237e022c", GitTreeState:"clean", BuildDate:"2020-02-21T10:02:50Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```
- OS (e.g. from `/etc/os-release`): `OSX`

**Comment by yuzisun** [comment]
@orezahc this should be fixed in 0.3.0 already

**Comment by orezahc** [comment]
@yuzisun thanks!
